File: app/src/main/java/com/example/popularmovies/ui/decorations/MarginItemDecoration.java
package com.example.popularmovies.ui.decorations;

import android.content.Context;
import android.graphics.Rect;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.DisplayMetrics;
import android.util.TypedValue;
import android.view.View;

public class MarginItemDecoration extends RecyclerView.ItemDecoration {
    private int marginLeft;
    private int marginRight;
    private int marginTop;
    private int marginBottom;

    public MarginItemDecoration(Context context, int marginLeft, int marginRight, int marginTop, int marginBottom) {
        DisplayMetrics metrics = context.getResources().getDisplayMetrics();
        this.marginLeft = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginLeft, metrics);
        this.marginRight = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginRight, metrics);
        this.marginTop = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginTop, metrics);
        this.marginBottom = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginBottom, metrics);
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        outRect.left = marginLeft;
        outRect.right = marginRight;
        outRect.bottom = marginBottom;
        outRect.top = marginTop;
    }

}

File: app/src/main/java/com/recursia/popularmovies/presentation/views/decorations/MarginItemDecoration.kt
package com.recursia.popularmovies.presentation.views.decorations

import android.content.Context
import android.graphics.Rect
import android.util.TypedValue
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class MarginItemDecoration(
        context: Context,
        marginLeft: Int,
        marginRight: Int,
        marginTop: Int,
        marginBottom: Int
) : RecyclerView.ItemDecoration() {
    private val marginLeft: Int
    private val marginRight: Int
    private val marginTop: Int
    private val marginBottom: Int

    init {
        val metrics = context.resources.displayMetrics
        this.marginLeft = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginLeft.toFloat(), metrics).toInt()
        this.marginRight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginRight.toFloat(), metrics).toInt()
        this.marginTop = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginTop.toFloat(), metrics).toInt()
        this.marginBottom = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, marginBottom.toFloat(), metrics).toInt()
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        outRect.left = marginLeft
        outRect.right = marginRight
        outRect.bottom = marginBottom
        outRect.top = marginTop
    }
}
